File: php/tests/Unit/Domain/Topology/ValueObject/LocationCodeTest.php
<?php
declare(strict_types=1);

namespace Tests\Unit\Domain\Topology\ValueObject;

use Alxarafe\App\Domain\Topology\ValueObject\LocationCode;
use InvalidArgumentException;
use PHPUnit\Framework\TestCase;

final class LocationCodeTest extends TestCase
{
    public function testCreate(): void
    {
        $code = new LocationCode('WH01-A01-01-01');
        self::assertSame('WH01-A01-01-01', $code->value());
    }

    public function testEmpty(): void
    {
        $this->expectException(InvalidArgumentException::class);
        new LocationCode('');
    }

    public function testTooLong(): void
    {
        $this->expectException(InvalidArgumentException::class);
        new LocationCode(str_repeat('A', 51));
    }

    public function testEquals(): void
    {
        $a = new LocationCode('WH01-A01-01-01');
        $b = new LocationCode('WH01-A01-01-01');
        self::assertTrue($a->equals($b));
    }

    public function testToString(): void
    {
        $code = new LocationCode('WH01-A01-01-01');
        self::assertSame('WH01-A01-01-01', (string) $code);
    }
}

File: java/src/test/java/com/alxarafe/domain/topology/valueobject/LocationCodeTest.java
package com.alxarafe.domain.topology.valueobject;

import com.alxarafe.app.domain.topology.valueobject.LocationCode;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class LocationCodeTest {

    @Test
    void create() {
        var code = new LocationCode("WH01-A01-01-01");
        assertEquals("WH01-A01-01-01", code.value());
    }

    @Test
    void nullValue() {
        assertThrows(NullPointerException.class, () -> new LocationCode(null));
    }

    @Test
    void blank() {
        assertThrows(IllegalArgumentException.class, () -> new LocationCode(""));
    }

    @Test
    void tooLong() {
        assertThrows(IllegalArgumentException.class, () -> new LocationCode("A".repeat(51)));
    }
}
